storage_account/main.tf

```
@@ -107,7 +107,13 @@ resource "azurerm_storage_account" "this" {
 # Enable advanced threat protection
 resource "azurerm_advanced_threat_protection" "this" {
   target_resource_id = azurerm_storage_account.this.id
-  enabled            = var.advanced_threat_protection
+  enabled            = var.advanced_threat_protection == true && var.use_legacy_defender_version == true
```

```suggestion
  enabled            = var.advanced_threat_protection == true || var.use_legacy_defender_version == true
```

this should be an OR?

---

Not really, because the `var.use_legacy_defender_version` enables either `azurerm_advanced_threat_protection` OR `azurerm_security_center_storage_defender`. Meanwhile, `var.advanced_threat_protection` enables `var.use_legacy_defender_version` value check

---

ok